### ISSUE TITLE
Terraform provider does not support BQ's native schema update behavior. Specifically remove column operations

### ISSUE THREAD

**Issue by red8888** [hashicorp/terraform-provider-google#11345]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### TLDR

The terraform provider should behave like native BQ does wrt schema updates: https://cloud.google.com/bigquery/docs/managing-table-schemas

It currently does not. Remove column operations are treated destructively- terraform wants to completely destroy and recreate the table.

### Terraform Version

Terraform v0.14.4
+ provider registry.terraform.io/hashicorp/google v3.67.0
+ provider registry.terraform.io/hashicorp/google-beta v3.67.0

### Affected Resource(s)

google_bigquery_table

### Terraform Configuration Files

```
# original config
resource "google_bigquery_table" "default" {
  dataset_id = google_bigquery_dataset.default.dataset_id
  table_id   = "bar"
  schema = <<EOF
[
  {
    "name": "removeme",
    "type": "BOOLEAN",
    "mode": "NULLABLE",
    "description": "I should be able to remove this"
  },
  {
    "name": "permalink",
    "type": "STRING",
    "mode": "NULLABLE",
    "description": "The Permalink"
  },
  {
    "name": "state",
    "type": "STRING",
    "mode": "NULLABLE",
    "description": "State where the head office is located"
  }
]
EOF
}
```

```
# remove column removeme and balk at terraform's unmitigated audacity
resource "google_bigquery_table" "default" {
  dataset_id = google_bigquery_dataset.default.dataset_id
  table_id   = "bar"
  schema = <<EOF
[
  {
    "name": "permalink",
    "type": "STRING",
    "mode": "NULLABLE",
    "description": "The Permalink"
  },
  {
    "name": "state",
    "type": "STRING",
    "mode": "NULLABLE",
    "description": "State where the head office is located"
  }
]
EOF
}
```

### Expected Behavior

I modify a schema to remove a column.

Terraform updates in place, just like it currently does for add column operations.

### Actual Behavior

I modify a schema to remove a column.

Terraform wants to destroy the entire table and recreate it.

### Steps to Reproduce

See ### Terraform Configuration Files
